streams_scraper/streams_scraper.py

```diff
@@ -35,84 +43,88 @@ def configure_driver():
-    
-    base_url = "https://www.google.com/search"
-    game_data = []
+def game_name(game):
+    return f"{game['home_team_name']} vs {game['away_team_name']}"
 
+
+def search_game_links(driver, future_games, games_today):
+    base_url = "https://www.bing.com/search"
+    stream_data = []
+
+    streams_number = 1
     for game in future_games:
         search_url = f"{base_url}?q={game.replace(' ', '+')}+watch"
         driver.get(search_url)
-        links_elements = driver.find_elements(By.CLASS_NAME, 'LC20lb')
+        links_elements = driver.find_elements(By.XPATH, '//h2/a')
         urls = [link.get_attribute('href') for link in links_elements[:3]]
 
         matching_game = next(
-            (g for g in games_today if f"{g['home_team_name']} vs {g['away_team_name']} {g['date']}" == game),
+            (g for g in games_today['items'] if f"{g['home_team_name']} vs {g['away_team_name']} {g['date']}" == game),
             None
         )
-        start_time = matching_game['time'] if matching_game else "Unknown"
-
-        sport_id = "unknown"
-        if matching_game:
-            sport_name = matching_game.get("sport_name", "unknown")
-            try:
-                sport_id = get_sport_index_by_name(session, sport_name)
-            except SportNotFoundError:
-                print(f"Sport '{sport_name}' not found in the database.")
-
-        game_info = {
-            "stream_url": urls[0] if urls else None,
-            "start_time": start_time,
-            "sport_id": sport_id
-        }
-        game_data.append(game_info)
-
-    return game_data
-
-
-def save_stream_data(session, game_data):
-    
-    streams_data = []
-
-    for game in game_data:
-        streams_data.append({
-            "stream_url": game["stream_url"],
-            "sport_id": game["sport_id"],
-            "start_time": game["start_time"]
-        })
 
-    save_json_stream_to_streams_table(session, streams_data)
+        if not matching_game:
+            continue
+
+        print(f"Matched stream #{streams_number} is being processed")
+        streams_number += 1
+
+        date = datetime.strptime(matching_game.get("date"), "%Y-%m-%d")
+        
+        stream_dto = StreamDTO(
+            title=game_name(matching_game),
+            stream_urls=urls if urls else None,
+            start_time=date,
+            sport_id=matching_game.get("sport_id")
+        )
+        stream_data.append(stream_dto)
+
+    return stream_data
+
+
+@inject
+def save_stream_data(streams_data):
+    stream_service.save_streams_to_streams_table(streams_data)
 
 
+@inject
 def main():
-    
-    session = SessionLocal()
     driver = configure_driver()
 
     try:
-        
-        
-        games_today = get_games_today(session)
-        future_games = filter_future_games(games_today)
-
-        
-        game_data = search_game_links(driver, future_games, games_today, session)
+        dto = SearchDTO().load({
+            "filters": [
```

Comment:
There must be a way of creating these objects without this string-dancing shit...